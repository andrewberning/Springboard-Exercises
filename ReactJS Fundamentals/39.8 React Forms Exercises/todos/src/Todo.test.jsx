import { expect, it, vi } from 'vitest';
import { render, fireEvent } from "@testing-library/react"
import '@testing-library/jest-dom';
import Todo from './Todo';

it("renders without crashing", function() {
  render(<Todo />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when editing", function() {
  const { asFragment, getByText } = render(<Todo />);
  fireEvent.click(getByText("Edit"));
  expect(asFragment()).toMatchSnapshot();
});

it("runs the update function on form submit", function() {
  const updateMock = vi.fn();
  const { getByText } = render(<Todo update={updateMock} />);
  fireEvent.click(getByText("Edit"));
  fireEvent.click(getByText("Update"));
  expect(updateMock).toHaveBeenCalled();
});

it("runs the delete function on button click", function() {
  const removeMock = vi.fn();
  const { getByText } = render(<Todo remove={removeMock}/>)
  fireEvent.click(getByText("X"));
  expect(removeMock).toHaveBeenCalled();
});