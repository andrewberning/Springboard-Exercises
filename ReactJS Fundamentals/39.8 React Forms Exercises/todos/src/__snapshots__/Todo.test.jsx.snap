// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <li>
      default todo
    </li>
    <button>
      Edit
    </button>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when editing 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        type="text"
        value="default todo"
      />
      <button>
        Update
      </button>
    </form>
  </div>
</DocumentFragment>
`;
